spec/controllers/pulse_spec.rb

```diff
@@ -1,31 +1,89 @@
 require 'spec_helper'
 require 'routemaster/controllers/pulse'
 require 'spec/support/rack_test'
+require 'spec/support/persistence'
+require 'spec/support/env'
 
 describe Routemaster::Controllers::Pulse, type: :controller do
   let(:app) { described_class }
 
-  let(:perform) { get '/pulse' }
+  describe 'GET /pulse' do
+    let(:perform) { get '/pulse' }
 
-  context 'when all is fine' do
+    context 'when all is fine' do
+      it 'succeeds' do
+        perform
+        expect(last_response.status).to eq(204)
+      end
 
-    it 'succeeds' do
-      perform
-      expect(last_response.status).to eq(204)
+      it 'does not return anything' do
+        perform
+        expect(last_response.body).to be_empty
+      end
     end
 
-    it 'does not return anything' do
-      perform
-      expect(last_response.body).to be_empty
+    context 'when the pulse service fails' do
+      before { allow_any_instance_of(Routemaster::Services::Pulse).to receive(:run).and_return(false) }
+
+      it 'returns 500' do
```

Comment:
I haven't modified this logic - this is the preexisting behaviour, so a bit out of scope here.
We could provide a better message, but given this is the health-check endpoint you wouldn't be able to see it anywhere?